app/components/UI/Ramp/Deposit/Views/ProviderWebview/ProviderWebview.tsx

```
@@ -25,49 +29,117 @@ export const createProviderWebviewNavDetails =
-    if (!userDetailsResponse) return 'No KYC data available';
+  useEffect(() => {
+    const fetchPaymentUrl = async () => {
+      if (!ottResponse) {
+        return;
+      }
+
+      await generatePaymentUrl(ottResponse.token, quote, selectedAddress);
+    };
+
+    fetchPaymentUrl();
+  }, [ottResponse, generatePaymentUrl, quote, selectedAddress]);
+
+  const handleNavigationStateChange = (navState: { url: string }) => {
+    if (navState.url.startsWith('https://metamask.io')) {
+      try {
+        const urlObj = new URL(navState.url);
+        const orderId = urlObj.searchParams.get('orderId');
 
-    const status = userDetailsResponse?.kyc?.l1?.status;
-    const type = userDetailsResponse?.kyc?.l1?.type;
-    return `The user KYC is ${status || 'unknown'} for the type ${
-      type || 'unknown'
-    }`;
+        if (orderId) {
+          navigation.navigate(
+            ...createOrderProcessingNavDetails({
+              quoteId: orderId,
+            }),
+          );
+        }
+      } catch (e) {
+        console.error('Error extracting orderId from URL:', e);
+      }
+    }
   };
 
+  const error = ottError || webviewError || paymentUrlError;
+
+  if (error) {
+    return (
+      <ScreenLayout>
+        <ScreenLayout.Body>
+          <ErrorView
+            description={error}
+            ctaOnPress={() => {
+              setKey((prevKey) => prevKey + 1);
+              setWebviewError('');
+            }}
+            location="Provider Webview"
+          />
+        </ScreenLayout.Body>
+      </ScreenLayout>
+    );
+  }
+
+  if (paymentUrl) {
+    return (
+      <ScreenLayout>
+        <ScreenLayout.Body>
+          <WebView
+            key={key}
+            source={{ uri: paymentUrl }}
+            onNavigationStateChange={handleNavigationStateChange}
+            onHttpError={(syntheticEvent) => {
+              const { nativeEvent } = syntheticEvent;
+              if (nativeEvent.url === paymentUrl) {
```

what if these are not equal?

---

The error is silently ignored. This behavior is inherited from the Checkout WebView handler: https://github.com/MetaMask/metamask-mobile/blob/d7a48acb86a0949fe58a2ac4eb543cd46b7e9ab7/app/components/UI/Ramp/Aggregator/Views/Checkout.tsx#L182